pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -101,70 +100,80 @@ func (p RESTStorageProvider) GroupName() string {
 
 // PostStartHook returns the hook func that launches the config provider
 func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
-	return PostStartHookName, func(hookContext genericapiserver.PostStartHookContext) error {
-		flowcontrolClientSet := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+	return PostStartHookName, ensureAPFBootstrapConfiguration(), nil
+
+}
+
+func ensureAPFBootstrapConfiguration() genericapiserver.PostStartHookFunc {
+	return func(hookContext genericapiserver.PostStartHookContext) error {
+		clientset, err := flowcontrolclient.NewForConfig(hookContext.LoopbackClientConfig)
+		if err != nil {
+			return fmt.Errorf("failed to initialize clientset for APF - %w", err)
+		}
+
+		// get a derived context that gets cancelled after 30s or
+		// when the StopCh gets closed, whichever happens first.
+		ctx, cancel := wait.ContextFromChannelAndMaxWaitDuration(hookContext.StopCh, 30*time.Second)
```

Comment:
- it will try for `30s` to initialize (copied it from RBAC post start hook) let me know if this is okay.

- i also added a new function in apimachinery `wait` package  `PollImmediateUntilWithContext`, let me know if this is okay to keep or you want this in a new PR.

Reply:
I'm not convinced if 30s isn't too low, but other than that the pattern looks ok.

I already commented on the wait part - let's move it to a separate PR.